src/main/java/storage/Storage.java

```diff
@@ -139,35 +144,15 @@ public AuthorList readAuthorListFromDataFile() {
      */
     public void saveAuthorListToDataFile(AuthorList authorList) {
         assert authorList != null : "authorList cannot be null";
-        try (FileWriter writer = new FileWriter(dataFile)) {
-            gson.toJson(authorList, writer);
-            logger.info("Data saved");
-        } catch (IOException e) {
-            logger.warning("Problems saving file, data will not be saved!" + e.getMessage());
-        }
-    }
-}
-
-/**
- * This package private class defines a custom <code>Gson</code> serializer for the <code>Author</code> class because
- * the native one cannot handle the circular reference (bidirectional navigability) between an <code>Author</code>
- * and their <code>Manga</code>. Specifically, it manually sets the <code>author</code> of each <code>Manga</code>
- * instance that is deserialized.
- */
-class AuthorDeserializer implements JsonDeserializer<Author> {
-    @Override
-    public Author deserialize(JsonElement json, Type typeOfAuthor, JsonDeserializationContext context)
-            throws JsonParseException {
-        JsonObject jsonObject = json.getAsJsonObject();
-        String authorName = jsonObject.get("authorName").getAsString();
-        Author author = new Author(authorName);
-
-        JsonArray mangaArray = jsonObject.getAsJsonArray("mangaList");
-        for (JsonElement mangaElement : mangaArray) {
-            Manga manga = context.deserialize(mangaElement, Manga.class);
-            manga.setAuthor(author);    // manually sets the author
-            author.addManga(manga);
+        boolean hasIssuesWithDataFile = createFileIfNeeded();
+        if (hasIssuesWithDataFile) {
```

Comment:
Minor semantic issue. `createFileIfNeeded` returns `false` if created successfully aka no issue. The `if` branch runs when `hasIssuesWithDataFile` is true. What was the intended flow here? If `hasIssues`, we try to write? If we know that there are issues, why are we still trying to write? Correct me if I am missing something. Might be better to update variable naming and method flow to reflect the obvious and 'happy' path.

Reply:
My bad, I was supposed to invert the flag here. I initially named it as `hasNoIssuesWithDataFile`, but decided against it because of the negative modifier in the name.